examples/01_plotting/plot_3d_map_projection.py

```diff
@@ -0,0 +1,38 @@
+"""
+Making a surface plot of a 3D statistical map
+=============================================
+
+Download a statistical map from Neurovault and a cortical mesh (fsaverage),
+project the 3D data on the mesh and display a surface plot.
+
+"""
+
+from nilearn import datasets
+from nilearn import plotting
+
+##############################################################################
+# Get a statistical map
+# ---------------------
+
+brainpedia = datasets.fetch_neurovault_ids(image_ids=(32015,))
+image = brainpedia.images[0]
+
+##############################################################################
+# Get a cortical mesh
+# -------------------
+
+fsaverage = datasets.fetch_surf_fsaverage5()
+
+##############################################################################
+# Sample the 3D data around each node of the mesh
+# -----------------------------------------------
+
+texture = plotting.niimg_to_surf_data(image, fsaverage.pial_left)
```

Comment:
I hit nans while using faverage.infl_left or infl_right. How can we overcome this problem in ```niimg_to_surf_data``` ?
Keeping them to zero works ?

Reply:
- infl_left and infl_right are the inflated surfaces: the nodes of the mesh have
  been moved to make the surface of the cortex smoother, inflated for display.
  They are not any more at their original position in the cortex, so it doesn't
  really make sense to sample the volume at these positions. I think what people
  would do is: use pial_left and pial_right, the true positions of the cortical
  mesh, to get the projectin, then use infl_left and infl_right, along with the
  computed projection, to make the display.

- for now, we get nans for vertices that are outside the image, or outside the
  mask if one is provided. I also don't really like returning nans, but values
  at these nodes truly aren't defined. we could provide a fill value kwarg?

Reply:
Surface plotting users know exactly which mesh to use to get right display. You are right nans are problem. But keeping them zero is also not valid. What would be your choice of filling value ?

Reply:
I don't think there is any reasonable default for filling value except nan, but we could allow the user to  choose a different one. If we use anything else than nan, e.g. 0, the user will not now if the image intensity at this point was 0, or it was filled because outside of mask or image